internal/bminventory/inventory.go

```diff
@@ -1845,22 +1845,33 @@ func (b *bareMetalInventory) updateNonDhcpNetworkParams(updates map[string]inter
 	}
 
 	var err error
-	*machineCidr, err = network.CalculateMachineNetworkCIDR(apiVip, ingressVip, cluster.Hosts)
+	err = verifyParsableVIPs(apiVip, ingressVip)
 	if err != nil {
-		log.WithError(err).Errorf("failed to calculate machine network cidr for cluster: %s", params.ClusterID)
-		return common.NewApiError(http.StatusBadRequest, err)
+		log.WithError(err).Errorf("Failed validating VIPs of cluster id=%s", params.ClusterID)
+		return err
 	}
+	*machineCidr, _ = network.CalculateMachineNetworkCIDR(apiVip, ingressVip, cluster.Hosts)
```

Comment:
If the vips are not empty and there are no hosts, it seems the function returns with an error.  Is this the reason to ignore?

Reply:
yes

Reply:
Still, I think errors should not be ignored.   There is another possible error there that the IP is illegal

Reply:
That error that we don't want to miss moved to another function a few lines above  `err = verifyParsableVIPs(apiVip, ingressVip)`

I didn't delete the error from the original function since it also been used in other places

Reply:
I don't think we should have code where we ignore errors.  It will be better to skip this function altogether in this case.

Reply:
In addition, there is a case that vips are set in this location but there are no hosts.  Didn't see any location where the machine-cidr is set after hosts have been discovered.
@filanov FYI

Reply:
Isn't it part of the async validation?

Reply:
> > > > Just making sure, is it relevant if we are working with VIP DHCP allocation = false? @ori-amizur ?
> > > 
> > > 
> > > @filanov what is relevant?
> > 
> > 
> > VIP DHCP allocation is not exposed in kube-api, we set it to false when creating the cluster.
> > So in this case we don't need to set machine-network cidr right?
> 
> Still don't understand what is not relevant.
> Anyway VIP DHCP allocation is false machine-cidr should not be set, unless (for example), we use single node installation.

OK i think that i understand the issue, because we allow to set api-vip and ingress vip without hosts then no function will called to set machine-network-cidr after hosts are discovered?

Reply:
We don't update machine-cidr during validations.

Reply:
the calculation will be added to the cluster monitor and will be removed from here